src/tests/ngraph_helpers/snippets_ngraph_functions/include/precision_propagation_function.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <ngraph/ngraph.hpp>
+#include "ngraph/opsets/opset1.hpp"
+#include "snippets/op/convert_saturation.hpp"
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+class PrecisionPropagationFunction {
+public:
+    template<typename T>
+    static std::shared_ptr<ngraph::Function> get(
+        const ngraph::element::Type precision1,
+        const ngraph::PartialShape& inputShape1,
+        const ngraph::element::Type precision2,
+        const ngraph::PartialShape& inputShape2,
+        const ngraph::element::Type constant_precision,
+        const std::pair<element::Type, element::Type>& convertion_before_op1 = std::pair<element::Type, element::Type>(),
+        const element::Type convertion_before_op2_1 = element::undefined,
+        const std::pair<element::Type, element::Type>& convertion_before_op2_2 = std::pair<element::Type, element::Type>(),
+        const element::Type convertion_after_op2 = {}) {
+        const auto branch1 = make_branch(precision1, inputShape1, 1, convertion_before_op1.first);
+        const auto branch2 = make_branch(precision2, inputShape2, 2, convertion_before_op1.second);
+
+        std::shared_ptr<Node> parent = std::make_shared<T>(branch1.second, branch2.second);
+        parent->set_friendly_name("add");
+
+        parent = create_convert(parent, convertion_before_op2_1);
+
+        const auto maximum_in2_type = convertion_before_op2_2.second == element::undefined ?
+            constant_precision :
+            convertion_before_op2_2.second;
+        if ((convertion_before_op2_2.first == element::undefined) &&
+            (parent->get_output_element_type(0) != maximum_in2_type)) {
+            parent = std::make_shared<ngraph::snippets::op::ConvertSaturation>(parent, maximum_in2_type);
+        }
+
+        parent = std::make_shared<ngraph::opset1::Maximum>(
+            create_convert(parent, convertion_before_op2_2.first),
+            create_convert(
+                std::make_shared<ngraph::opset1::Constant>(constant_precision, Shape{}, std::vector<float>{0.f}),
+                convertion_before_op2_2.second));
+        parent->set_friendly_name("maximum");
+
+        parent = create_convert(parent, convertion_after_op2);
+
+        const auto result = std::make_shared<ngraph::opset1::Result>(parent);
+        result->set_friendly_name("result");
+
+        const ngraph::ResultVector results{result};
+        const ngraph::ParameterVector parameters{branch1.first, branch2.first};
+        const auto model = std::make_shared<ngraph::Function>(results, parameters, "SnippetsPrecisionPropagation");
+        model->validate_nodes_and_infer_types();
```

Comment:
Do we really need to validate model just after creation? I suppose that `ov::Model` must call nodes validation at the construction stage